priiloader/source/hacks.cpp

```diff
@@ -549,14 +563,62 @@ s8 LoadSystemHacks(bool load_nand)
 
 			mem_free(fbuf);
 
-			//Set all system_hacks for system menu > max_version || sysver < min_version to 0
+			
 			unsigned int sysver = GetSysMenuVersion();
+			
+			// Make sure that all master hacks are disabled in the state array. 
+			// That way, when all sub-hacks are disabled, the master hack will 
+			// be disabled as well. Then, in the next loop, if a sub hack is 
+			// selected, the master hack will be enabled again. 
+			for( u32 i=0; i < system_hacks.size(); ++i)
+			{
+				if (system_hacks[i].provide.length() > 0) {
+					// This is a master hack
+					states_hash[i] = 0; 
+				}
+			}
+			
+			
+			//Set all system_hacks for system menu > max_version || sysver < min_version to 0
 			for( u32 i=0; i < system_hacks.size(); ++i)
 			{
 				if( system_hacks[i].max_version < sysver || system_hacks[i].min_version > sysver)
 				{
 					states_hash[i] = 0;
 				}
+				
+				// If there is a hack that requires another one, make sure that that is present: 
+				else if (system_hacks[i].require.length() > 0 )  {

```

Comment:
though i have no experience with them in C++, this can probably be easily rewritten into a lambda expression

for example : https://stackoverflow.com/questions/15517991/search-a-vector-of-objects-by-object-attribute

Reply:
A lambda expression? What would be the point? The existing code is already looping through the list anyways, why add another lambda expression in there instead of just checking the length of each element's require parameter?

Reply:
Also, in my opinion, a simple for loop is way more readable than one of these lambda expressions. Although that might just be because I don't have much experience in C++, I mainly use C.

Reply:
i think it is an experience thing. i could read the example i linked just fine. i am however used to them from my experience in C# ( + LINQ ) & javascript . i had to look for a sec to see what is what, but it was the first time looking at a lambda expression in C++.
i do agree in c++ they can get very confusing/complex but the example i linked should work fine and is straight to the point and will work faster

i just did a test on the wii btw (looking through a list of 100 items for an item with a specific description which is in the middle of the list). the lambda took 1/3th less time.
the further down the list i pushed what it was looking for, the more lambda was crushing it.
100 items, 99th is the item we are looking for -> 10x faster.
100 items, 1st item is the item -> same time between both.

yes it is going to loop through the list, but in this case (looking for an item) lambda is a lot less taxing on the system, even on the powerpc.

also, this is C/C++, not java. java is far from good at performance anyway...